File: js연습/level2/귤고르기.js
function solution(k, tangerine) {
  const obj = {};
  tangerine.forEach((n) => {
    obj[n] = ++obj[n] || 1
  })
  let i = 0;
  let answer = 0;
  let arr = Object.values(obj).sort((a, b) => b-a);
  for (o of arr) {
    i += o;
    answer += 1
    if (i >= k) {
      break
    }
  }
  return answer;
}

File: 프로그래머스 문제/level2/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    dic = {}

    for i in set(tangerine):
        dic[i] = 0

    for t in tangerine:
        dic[t] += 1
    dic = sorted(dic.items(), key=lambda x: x[1], reverse=True)

    for key, value in dic:
        k = k - value
        answer += 1
        if k <= 0:
            break
    return answer